### PR TITLE
refactor(core,schemas,console): optimize the sso connector endpoints naming

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

This PR includes the following updates:

1. Rename the endpoint `/sso-connector-factories` to `/sso-connector-providers`. Align the naming convention. 
2. Update the response type of the `/sso-connector-providers`, endpoint as well.  Directly return a list of providers. Let the console to categorize the standard connector and enterprise connector. 
3. Update the openapi docs
4. Declare the `SAMLResponse` type in the request body of the SAML athn ACS endpoint. 

## Testing

image

image

image

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [x] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: .changeset/calm-ladybugs-doubt.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@logto/core": patch
+---
+
+add summary and description to APIs
```
Comment (charIeszhao) [inline .changeset/calm-ladybugs-doubt.md:5]: nit: I think we should use sentence case in changelogs

File: packages/console/src/pages/EnterpriseSso/SsoCreationModal/index.tsx [hunk 2]
```diff
@@ -54,19 +55,22 @@ function SsoCreationModal({ isOpen, onClose: rawOnClose }: Props) {
...
+    () => categorizeSsoConnectorProviders(data),
+    [data]
+  );
 
   const radioGroupSize = useMemo(
-    () => getConnectorRadioGroupSize(standardConnectors.length + providerConnectors.length),
-    [standardConnectors, providerConnectors]
+    () => getConnectorRadioGroupSize(standardProviders.length + enterpriseProviders.length),
+    [standardProviders, enterpriseProviders]
   );
```
Comment (charIeszhao) [inline packages/console/src/pages/EnterpriseSso/SsoCreationModal/index.tsx:66]: nit: IMO we should revisit these two `useMemo` usages later. Seems unnecessary to me.

File: packages/core/src/routes/sso-connector/index.ts [hunk 3]
```diff
@@ -52,34 +51,21 @@ export default function singleSignOnRoutes<T extends AuthedRouter>(...args: Rout
...
-    '/sso-connector-factories',
+    '/sso-connector-providers',
     koaGuard({
-      response: ssoConnectorFactoriesResponseGuard,
+      response: ssoConnectorProvidersResponseGuard,
       status: [200],
     }),
     async (ctx, next) => {
       const { locale } = ctx;
       const factories = Object.values(ssoConnectorFactories);
```
Comment (charIeszhao) [inline packages/core/src/routes/sso-connector/index.ts:66]: Should we also rename these `factories` to `providers`?

Comment (simeng-li) [reply packages/core/src/routes/sso-connector/index.ts:66]: Internal code reference should be fine. Looks straight forward. Just need to hide the factories from the API/data level. factories more like our internal connector logic.

**Review by wangsijie — APPROVED** [review]

**Pull request merged**